Add FAQ advising against underscores in hostnames. (#1366)
OttoWinter/esphomedocs: guides/faq.rst

Always back up all your files!

> 

Reviewer: ```suggestion
.. _faq-device_name_underscores:
```

So we can link to it better

Reviewer: Nvm, apparently the link is not generated with that.

Why shouldn't I use underscores in my device name?
--------------------------------------------------

> The top level ``name:`` field in your .yaml file defines the node name(/hostname) on the local network.  According to `RFC1912 <https://datatracker.ietf.org/doc/html/rfc1912>`_, underscore characters (``_``) in hostnames are not valid.  In reality some local DNS/DHCP setups will be ok with underscores and some will not.  If connecting via a static IP address, there will probably be no issues.  In some cases, initial setup using an underscore works, but then the connection fails when Home Assistant restarts.  Recommendation: use hyphen (``-``) instead of underscore if you can.  

Reviewer 2: >  but then the connection fails when Home Assistant restarts.

could this be more like: "but then the connection **might** fail when Home Assistant restarts."

Author: Agreed, done.

> Important: follow these `instructions </components/esphome.html#changing-esphome-node-name>`_ to use the ``use_address`` parameter when renaming a live device, as the connection to an existing device will only work with the old name until the name change is complete.

Reviewer 2: ```suggestion
Follow these `instructions </components/esphome.html#changing-esphome-node-name>`_ to change the underscores in the name to hyphens.
```

No need to explain why here, it's explained in the instructions.

Author: I think it is better to give some idea why here.  Many will assume they can just rename it.
